scripts/test_perfmon.py

```diff
@@ -0,0 +1,112 @@
+        /dev/sda5        4054752 59820   3785220   2% /var/log
+        """
+        if cmd == "df /etc/passwd":
+            return df_etc_passwd
+        if cmd == "df /var/log":
+            return df_var_log
+        
+    def mock_subprocess_getoutput_same_file_system(self, cmd):
+        df_etc_passwd = r"""Filesystem     1K-blocks    Used Available Use% Mounted on
+        /dev/sda5       18402132 2244748  15213668  13% /
+        """
+        df_var_log = r"""Filesystem     1K-blocks  Used Available Use% Mounted on
+        /dev/sda5        4054752 59820   3785220   2% /var/log
+        """
+        if cmd == "df /etc/passwd":
+            return df_etc_passwd
+        if cmd == "df /var/log":
+            return df_var_log
+
+    def test_get_percent_log_fs_usage(self, mock_getoutput):
+        """Assert that get_percent_log_fs_usage returns as expected"""
+        mock_getoutput.side_effect = self.mock_subprocess_getoutput
+
+        expected_percentage = 0.02
+        test_percentage = perfmon.get_percent_log_fs_usage(None)
+        self.assertAlmostEqual(test_percentage, expected_percentage, 7)
+        
+    def test_get_percent_log_fs_usage_same_file_system(self, mock_getoutput):
+        """Test where /etc/passwd and /var/log are in the same filesystem"""
+        mock_getoutput.side_effect = self.mock_subprocess_getoutput_same_file_system
+
+        test_percentage = perfmon.get_percent_log_fs_usage(None)
+        self.assertTrue(math.isnan(test_percentage))
+        
+    def test_get_percent_fs_usage(self, mock_getoutput):
+        """Assert that get_percent_fs_usage returns as expected"""
+        mock_getoutput.side_effect = self.mock_subprocess_getoutput
+
+        expected_percentage = 0.13
+        test_percentage = perfmon.get_percent_fs_usage(None)
+        self.assertAlmostEqual(test_percentage, expected_percentage, 7)
+@unittest.skipIf(sys.version_info < (3, 0), reason="requires python3")
+@patch("perfmon.Variable.update")
+@patch("perfmon.get_percent_fs_usage")
+@patch("perfmon.get_percent_log_fs_usage")
+@patch("perfmon.get_percent_mem_usage")
+class TestVMMonitor(unittest.TestCase):   
+    def test_process_rrd_updates(self, mock_get_percent_fs_usage, mock_get_percent_log_fs_usage, 
+                                 mock_get_percent_mem_usage, mock_update):
+        uuid = 'e1ae3f5d-4c8b-4575-bbb8-2af7e8a2c31e'
+        perfmon.all_xmlconfigs = {'e1ae3f5d-4c8b-4575-bbb8-2af7e8a2c31e': '<config><variable><name value="fs_usage"/><alarm_trigger_level value="0.9"/><alarm_trigger_period value="60"/><alarm_auto_inhibit_period value="3600"/></variable><variable><name value="mem_usage"/><alarm_trigger_level value="0.95"/><alarm_trigger_period value="60"/><alarm_auto_inhibit_period value="3600"/></variable><variable><name value="log_fs_usage"/><alarm_trigger_level value="0.9"/><alarm_trigger_period value="60"/><alarm_auto_inhibit_period value="3600"/></variable></config>'}
```

Comment:
Is it possible to make this more diff-friendlly by avoiding overlong lines? I understand there are no spaces in this string. But maybe a similar case below could be improved.

Reply:
Thanks!
Code updated.